File: backend/app/schemas/analytics.py
from pydantic import BaseModel, Field


class AnalyticsPoint(BaseModel):
    date: str
    value: int = Field(ge=0)


class DashboardAnalyticsResponse(BaseModel):
    window_days: int
    accounts_created: list[AnalyticsPoint]
    campaigns_created: list[AnalyticsPoint]
    totals: dict[str, int]

File: frontend/src/types/analytics.ts
export interface AnalyticsPoint {
  date: string;
  value: number;
}

export interface DashboardAnalytics {
  window_days: number;
  accounts_created: AnalyticsPoint[];
  campaigns_created: AnalyticsPoint[];
  totals: {
    accounts: number;
    campaigns: number;
  };
}
